esmvalcore/preprocessor/_time.py

```diff
@@ -1292,3 +1298,381 @@ def resample_time(
         raise ValueError(
             f"Time coordinate {dates} does not contain {requested} for {cube}")
     return cube
+
+
+def _lin_pad(array: np.ndarray, delta: float, pad_with: int) -> np.ndarray:
+    """Linearly pad an array on both sides with constant difference."""
+    end_values = (array[0] - pad_with * delta, array[-1] + pad_with * delta)
+    new_array = np.pad(array, pad_with, 'linear_ramp', end_values=end_values)
+    return new_array
+
+
+def _guess_time_bounds(time_coord: DimCoord) -> None:
+    """Guess coordinates of time coordinate in-place."""
+    if time_coord.has_bounds():
+        return
+    try:
+        time_coord.guess_bounds()
+    except ValueError:  # coordinate has only 1 point
+        point = time_coord.points[0]
+        time_coord.bounds = [[point - 0.5, point + 0.5]]
```

Comment:
This will almost always be wrong, but maybe it doesn't matter?

Reply:
True, but you are right, it doesn't matter: this is only used to bin the exact local times into time intervals (in this case we assume 1hr), and will be removed at a later stage.